stats/lib/app.js

```diff
@@ -0,0 +1,151 @@
+import * as Sentry from '@sentry/node'
+import Fastify from 'fastify'
+import cors from '@fastify/cors'
+import urlData from '@fastify/url-data'
+
+import { withFilter } from './request-helpers.js'
+
+import {
+  fetchDailyDealStats,
+  fetchDailyParticipants,
+  fetchMinersRSRSummary,
+  fetchMonthlyParticipants,
+  fetchParticipantChangeRates,
+  fetchParticipantScheduledRewards,
+  fetchParticipantRewardTransfers,
+  fetchRetrievalSuccessRate,
+  fetchDealSummary,
+  fetchDailyRetrievalResultCodes,
+  fetchDailyMinerRSRSummary,
+  fetchDailyRetrievalTimings,
+  fetchDailyMinerRetrievalTimings,
+  fetchMinersTimingsSummary
+} from './stats-fetchers.js'
+
+import { addPlatformRoutes } from './platform-routes.js'
+
+/** @typedef {import('@filecoin-station/spark-stats-db').PgPools} PgPools */
+/** @typedef {import('./typings.js').DateRangeFilter} DateRangeFilter */
+/** @typedef {import('./typings.js').RequestWithFilter} RequestWithFilter */
+/** @typedef {import('./typings.js').RequestWithFilterAndAddress} RequestWithFilterAndAddress */
+/** @typedef {import('./typings.js').RequestWithFilterAndAddress} RequestWithFilterAndMinerId */
+
+/**
+ * @param {object} args
+ * @param {string} args.SPARK_API_BASE_URL
+ * @param {import('@filecoin-station/spark-stats-db').PgPools} args.pgPools
+ * @param {boolean} args.logger
+ * @returns
+ */
+export const createApp = ({
+  SPARK_API_BASE_URL,
+  pgPools,
+  logger
+}) => {
+  const app = Fastify({ logger })
+  Sentry.setupFastifyErrorHandler(app)
+
+  app.register(cors, {
+    origin: ['http://localhost:3000', 'app://-']
+  })
+  app.register(urlData)
+
+  app.get('/deals/daily', async (/** @type {RequestWithFilter} */ request, reply) => {
```

Comment:
What do you think about introducing `handlers.js` back or extract registering handlers into a separate function and leaving `createApp` function only for creating and configuring the `Fastify` app object and registering middlewares, etc?

Reply:
What would be the advantage of that?

Reply:
In that case `createApp` function would do only only thing -- creating the app. Also in my opinion code would be much cleaner. Plus we already are doing that for platform routes so it would make sense to do it for stat routes.

Reply:
It would do more than creating the app though. It would configure the app, and add plugins. To me, the app definition, with all routes that make sense, should be in one file. It also means you have to jump to less files to get to the often important part.

I want to give your suggestion a try however, and implemented it in 98c04bd. What do you think?

Reply:
ping @pyropy